### DISCUSSION TITLE
compute

### DISCUSSION THREAD

**Discussion by jschindler89** [brucefan1983/GPUMD#641] [Q&A]
Hi Zheyong,

I am using "compute" to calculate forces and energies.

So as an example, I tried to use "compute 0 100 10 temperature" as given in the command example on the website, but my simulation stops when it gets to reading the command. The last line on the output file is: "Compute space and/or time average of:
"

How can I fix this?

Jack

**Comment by brucefan1983** [comment]
Could you post `model.xyz`?

**Reply by jschindler89** [reply]
Hi Zheyong,

Here are two things I tried: with/without grouping information.
[model.zip](https://github.com/user-attachments/files/15802712/model.zip)

Thank you,

Jack

**Reply by jschindler89** [reply]
I have used "compute 2 1 1 temperature" to reflect that I chose group method 2.

**Comment by brucefan1983** [comment]
In `model_2.xyz`, there is no grouping method, so you cannot use any keyword requiring a gouping method. 

In `model.xyz`, there is only one grouping method (the `0`-th grouping method), and all the atoms are in group `0`. So you cannot use `compute 2 ...`. 

Could you first explain what exactly  you are studying?

**Reply by jschindler89** [reply]
Ah I see. so I am supposed to use "compute 0 1 1 temperature" for model.xyz?

I am trying to understand what I can do using compute. Trying to learn how things work. I am new to the method. Sorry for bothering you too much. 

Jack

**Reply by brucefan1983** [reply]
It is hard to give advice wihout knowing your goal. If you are studying heat transport using NEMD, you can go through this tutorial:

https://gpumd.org/tutorials/thermal_transport_nemd.html
(https://github.com/brucefan1983/GPUMD/tree/master/examples/04_Carbon_thermal_transport_nemd_and_hnemd/ballistic)

Here you can see how `compute temperature` is used.
